ecosystem_cicd_tools/new_cicd/actions.py

```diff
@@ -45,6 +48,18 @@ def upload_assets_to_release(assets, release_name, repository, **_):
             os.environ.get('CIRCLE_USERNAME', 'earthmant')
         )
 
+    time.sleep(180)
```

Comment:
Instead of waiting `180` seconds, let's do a loop. This can be as simple as:

```python
max_time = 180  # it should not take longer than 180 seconds.
min_time = 20  # It should definitely take longer than 20 seconds.
interval = 10  # We check every 10 seconds.
current = 0
while True:
    if current < 20:
        current += interval
        continue
    if max_time <= current:
        raise RuntimeError('Timed out waiting for marketplace plugin update.')
    if all_plugin_stuff_is_in_marketplace(plugin_name, plugin_version):
        break
    sleep(interval)
    current += interval
```